dialog.cpp

```diff
@@ -297,6 +299,16 @@ void Dialog::applyFilters()
         isOR = true;
     }
 
+    if(ui->FltrsWidget->downgradedChecked()) {
+        filter += (isOR? " OR " : " AND (") + QString("op='downgraded'");
+        isOR = true;
+    }
+
+    if(ui->FltrsWidget->reinstalledChecked()) {
+        filter += (isOR? " OR " : " AND (") + QString("op='reinstalled'");
+        isOR = true;
+    }
+
     if(isOR)
         filter += ")";
     else
```

Comment:
The regex in `Dialog::readPacmanLogFile()` already ensures that one of the operation strings is present: `(installed|removed|upgraded|downgraded|reinstalled)`. So this else and the next line `filter += "AND (op!='installed' AND op!='upgraded' AND op!='removed')";` can be safely removed. Currently unchecking all checkboxes shows downgraded and reinstalled packages.